### PR TITLE
Sumer upgrade substrate version to 2.0.1

### PR DESCRIPTION
**Pull request by conectado**
This PR ports the changes upgrading substrate to v2.0.1 from antioch.

There should be no conflict due to this changes with antioch, however some exists where sumer diverged from antioch on unrelated ways.

### CODE REVIEW

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -70,8 +84,8 @@ pub use content::MaxNumber;
 pub const VERSION: RuntimeVersion = RuntimeVersion {
     spec_name: create_runtime_str!("joystream-node"),
     impl_name: create_runtime_str!("joystream-node"),
-    authoring_version: 7,
-    spec_version: 16,
+    authoring_version: 9,
```
Comment (shamil-gadelshin) [inline runtime/src/lib.rs:87]: The spec version should be bumped further. Antioch has v9.3.0

File: runtime/Cargo.toml [hunk 2]
```diff
@@ -4,60 +4,60 @@ edition = '2018'
 name = 'joystream-node-runtime'
 # Follow convention: https://github.com/Joystream/substrate-runtime-joystream/issues/1
 # {Authoring}.{Spec}.{Impl} of the RuntimeVersion
-version = '7.16.0'
+version = '9.0.0'
```
Comment (shamil-gadelshin) [inline runtime/Cargo.toml]: The spec version should be bumped further. Antioch has v9.3.0

File: runtime-modules/proposals/engine/src/lib.rs [hunk 3]
```diff
@@ -1,6 +1,6 @@
 //! # Proposals engine module
 //! Proposals `engine` module for the Joystream platform. Version 2.
-//! The main component of the proposals system. Provides methods and extrinsics to create and
+//! The main component of the proposals frame_system. Provides methods and extrinsics to create and
```
Comment (shamil-gadelshin) [inline runtime-modules/proposals/engine/src/lib.rs]: Fix incorrect replace of the `system`

File: runtime-modules/proposals/discussion/src/lib.rs [hunk 4]
```diff
@@ -1,6 +1,6 @@
 //! # Proposals discussion module
 //! Proposals `discussion` module for the Joystream platform. Version 2.
-//! It contains discussion subsystem of the proposals.
+//! It contains discussion subframe_system of the proposals.
```
Comment (shamil-gadelshin) [inline runtime-modules/proposals/discussion/src/lib.rs]: Fix incorrect replace of the `system`

File: runtime-modules/proposals/codex/src/lib.rs [hunk 5]
```diff
@@ -1,10 +1,10 @@
 //! # Proposals codex module
 //! Proposals `codex` module for the Joystream platform. Version 2.
-//! Component of the proposals system. It contains preset proposal types.
+//! Component of the proposals frame_system. It contains preset proposal types.
```
Comment (shamil-gadelshin) [inline runtime-modules/proposals/codex/src/lib.rs]: Fix incorrect replace of the `system`

File: runtime-modules/proposals/codex/src/lib.rs [hunk 6]
```diff
@@ -1,10 +1,10 @@
...
 //!
 //! ## Overview
 //!
-//! The proposals codex module serves as a facade and entry point of the proposals system. It uses
+//! The proposals codex module serves as a facade and entry point of the proposals frame_system. It uses
```
Comment (shamil-gadelshin) [inline runtime-modules/proposals/codex/src/lib.rs]: Fix incorrect replace of the `system`

File: node/Cargo.toml [hunk 7]
```diff
@@ -3,7 +3,7 @@ authors = ['Joystream contributors']
 build = 'build.rs'
 edition = '2018'
 name = 'joystream-node'
-version = '3.7.2'
```
Comment (shamil-gadelshin) [inline node/Cargo.toml:6]: Please, bump version.

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]
Looks OK. Left a couple of issues. The important one - change runtime and node versions.

I bumped the runtime version locally and tested different configurations:
- runtime upgrade from antioch to sumer works both in dev and 'two-validators' modes
- upgraded sumer node works both in dev and 'two-validators' modes

However, when injecting the keys for the Sumer 'two-validators' running nodes. The chain starts producing blocks but without finalization. It resumes finalization after the restart. Is it a problem? @mnaamani

**Comment by shamil-gadelshin** [comment]
What compiler version do we use for this release? @conectado Is it the same as for Antioch (`nightly-2021-03-24`)?

**Comment by conectado** [comment]
> What compiler version do we use for this release? @conectado Is it the same as for Antioch (`nightly-2021-03-24`)?

Yes, I set that up using `setup.sh` which is the same as in antioch

**Comment by conectado** [comment]
> Looks OK. Left a couple of issues. The important one - change runtime and node versions.
> 
> I bumped the runtime version locally and tested different configurations:
> 
>     * runtime upgrade from antioch to sumer works both in dev and 'two-validators' modes
> 
>     * upgraded sumer node works both in dev and 'two-validators' modes
> 
> 
> However, when injecting the keys for the Sumer 'two-validators' running nodes. The chain starts producing blocks but without finalization. It resumes finalization after the restart. Is it a problem? @mnaamani

I ran the nodes using `./tests/network-tests/run-local-node-tests.sh`. But I'm not sure how to execute the runtime upgrade locally to tests if this works.

**Comment by mnaamani** [comment]
> However, when injecting the keys for the Sumer 'two-validators' running nodes. The chain starts producing blocks but without finalization. It resumes finalization after the restart. Is it a problem? @mnaamani

I had no issues with finalization running a `local` chain with alice and bob as validators

**Review by mnaamani — APPROVED** [review]
Will continue work to update in new PR

**Pull request merged**